.cirrus.yml

```
@@ -20,6 +20,8 @@ freebsd_task:
   pythoninfo_script:
     - cd build
     - make pythoninfo
+  bundle_ensurepip_script:
```

```suggestion
  # Download wheels for the venv step
  bundle_ensurepip_script:
```

Is this the reason for adding it here?

---

Anything using venv or ensurepip during testing needs this provisioning step. If there's no uses, it'd be unnecessary.

---

The actual reason to include it is that Cirrus CI failed without this and passed with it. I don't know why Free BSD requires the venv step though, as all other tests pass normally -- only documentation and hypothesis (which use venvs) seem to have an explicit venv step.

A

---

@AA-Turner It's not because of `venv` but `ensurepip` itself. The call chain looks as follows:
1. `test.test_tools.test_freeze.TestFreeze.test_freeze_simple_script` calls a CPython build preparation helper @ https://github.com/python/cpython/blob/cbb3a6f/Lib/test/test_tools/test_freeze.py#L28
2. That, in turn, calls `make install` here https://github.com/python/cpython/blob/cbb3a6f/Tools/freeze/test/freeze.py#L184.
3. `make install` calls `ensurepip` at https://github.com/python/cpython/blob/cbb3a6f/Makefile.pre.in#L1932-L1933 to provision it into the new Python install directory.
4. `ensurepip` uses the bundled pip wheel to run bootstrapping so it attempts to unpack it and freaks out when there's no dist for it: https://github.com/python/cpython/blob/cbb3a6f/Lib/ensurepip/__init__.py#L176.

---

> why Free BSD requires

It's not that FreeBSD or Cirrus requires the test explicitly, it's rather that this test is skipped on most platforms via https://github.com/python/cpython/blob/cbb3a6f/Lib/test/test_tools/test_freeze.py#L11-L20.